matborowiak-tutorial/part two (#7)
gatsbyjs/gatsby-pl: docs/tutorial/part-two/index.md

-->

> Witaj w drugiej części tutorialu Gatsby

Reviewer: ```suggestion
Witaj w drugiej części poradnika Gatsby
```
jak dla mnie _poradnik_ brzmi lepiej, _tutorialu_ trochę za bardzo polinglishem daje

Reviewer 2: Zgadzam się, poradnik albo samouczek. Myślę, że można to dodać to tableki w readme, żebyśmy używali tych samych sformułowań.

Reviewer 3: W 'part 0' @maciekgrzybek użył słowa `samouczek` jako spolszczenie `tutorial`. To może być równie dobra alternatywa.

Author: I `samouczek` i  `poradnik` brzmią lepiej od tutorial, ale z dwojga dla mnie samouczek brzmi trochę "swojsko", a poradnik jakoś tak bardziej oficjalnie, dlatego skłaniam się ku `poradnikowi`

> ## Czego naucysz się w tym tutorialu?

Reviewer: ```suggestion
## Czego nauczysz się w tym poradniku?
```

> W tej części zapoznasz się z możliwościami stylizacji stron internetowych Gatsby i przyjrzysz się bliżej zastosowaniu komponentów React'a w celu tworzenia witryn.

Reviewer 2: ```suggestion
W tej części zapoznasz się z możliwościami stylowania stron internetowych Gatsby i przyjrzysz się bliżej zastosowaniu komponentów React'a w celu tworzenia witryn.
```
Brzmi chyba sensowniej.

## Zastosowanie stylów globalnych

> Każda strona posiada jakiś rodzaj globalnego stylu. Zaliczają się do tego rzeczy takie jak typografia i kolory tła. Style te odpowiadają za ogólny odbiór i odczucie strony internetowej - zupełnie jak kolor i faktura ścian w pokoju definiuje panującą w nim atmosferę.

Reviewer: ```suggestion
Każda strona posiada jakieś style globalne. Zaliczają się do tego rzeczy takie jak typografia i kolory tła. Style te odpowiadają za ogólny odbiór i odczucie strony internetowej - zupełnie jak kolor i faktura ścian w pokoju definiuje panującą w nim atmosferę.
```

> ### Tworzenie stylów globalnych przy użyciu standarwodych plików CSS

Reviewer: ```suggestion
### Tworzenie stylów globalnych przy użyciu standardowych plików CSS
```

Author: Zastanawiam się nad kapitalizacją liter w takich terminach jak Style Globalne, Moduły CSS itp.  Wydaje mi się że takie wizualne odróżnienie pewnej terminologii od zwykłych słów budujących zdanie, może pomóc łatwiej zrozumieć taką dokumentację. Co myślicie? @lekterable @maciekgrzybek

Reviewer 2: Brzmi sensownie

Author: ```suggestion
### Tworzenie Stylów Globalnych przy użyciu standardowych plików CSS
```

Reviewer 2: @lekterable ?

Reviewer: 👍

> Jednym z najbardziej podstawowych sposobów aby dodać style do strony jest użycie globalnego arkusza styli `.css`.

Author: ```suggestion
Jednym z najbardziej podstawowych sposobów aby dodać style do strony jest użycie globalnego arkusza stylów `.css`.
```

> Jednym z najbardziej podstawowych sposobów aby dodać style do strony jest użycie globalnego stylesheet'u `.css`.

Reviewer: ```suggestion
Jednym z najbardziej podstawowych sposobów aby dodać style do strony jest użycie globalnego arkusza styli `.css`.
```
to w sumie luźna sugestia, @maciekgrzybek co myślisz?

Reviewer 2: Tu teź się zgadzam. Czasem warto zostawić słowo po angielsku bo w wielu wypadkach tłumaczenie nie ma sensu, ale sądze, że musimy się starać tłumaczyć jak najwięcej. 
P.S. Kolejne słówko do tabelki.

> Jednym z najbardziej podstawowych sposobów aby dodać style do strony jest użycie globalnego stylesheet'u `.css`.

Reviewer 2: ```suggestion
Jednym z najbardziej podstawowych sposobów na dodanie styli do strony, jest użycie globalnego stylesheet'u `.css`.
```

#### ✋ Utwórz nową stronę Gatsby

> Rozpocznij poprzez utworzenie nowej strony Gatsby. Najlepiej (szczególnie gdy używasz linii komend od niedawna) zamknąć wszystkie okna terminala, których używałeś w [części pierwszej](/tutorial/part-one/) i otwórz nowe okno dla drugiej części tutorialu.

Reviewer: ```suggestion
Rozpocznij poprzez utworzenie nowej strony Gatsby. Najlepiej (szczególnie gdy używasz linii komend od niedawna) zamknąć wszystkie okna terminala, których używałeś w [części pierwszej](/tutorial/part-one/) i otwórz nowe okno dla drugiej części poradnika.
```

> Rozpocznij poprzez utworzenie nowej strony Gatsby. Najlepiej (szczególnie gdy używasz linii komend od niedawna) zamknąć wszystkie okna terminala, których używałeś w [części pierwszej](/tutorial/part-one/) i otwórz nowe okno dla drugiej części tutorialu.

Reviewer 2: ```suggestion
Rozpocznij poprzez utworzenie nowej strony Gatsby. Najlepiej (szczególnie gdy używasz interfejsu wiersza poleceń od niedawna) zamknij wszystkie okna terminala, których używałeś w [części pierwszej](/tutorial/part-one/) i otwórz nowe okno dla drugiej części tutorialu.
```

> Rozpocznij poprzez utworzenie nowej strony Gatsby. Najlepiej (szczególnie gdy używasz interfejsu wiersza poleceń od niedawna) zamknij wszystkie okna terminala, których używałeś w [części pierwszej](/tutorial/part-one/) i otwórz nowe okno dla drugiej części tutorialu.

Author: ```suggestion
Rozpocznij poprzez utworzenie nowej strony Gatsby. Najlepiej (szczególnie gdy używasz interfejsu wiersza poleceń od niedawna) zamknij wszystkie okna terminala, których używałeś w [części pierwszej](/tutorial/part-one/) i otwórz nowe okno dla drugiej części poradnika.
```

Otwórz nowe okno terminala, stwórz nową stronę Gatsby "hello world" oraz uruchom serwer developerski (development server):

```shell
gatsby new tutorial-part-two https://github.com/gatsbyjs/gatsby-starter-hello-world
cd tutorial-part-two
```

> Utworzyłeś teraz nową stonę Gatsby (na podstawie gotowego projektu startowego "hello world") o następującej strukturze:

Reviewer: ```suggestion
Utworzyłeś teraz nową stronę Gatsby (na podstawie gotowego projektu startowego "hello world") o następującej strukturze:
```

Author: ```suggestion
Utworzyłeś teraz nową stronę Gatsby (na podstawie gotowego Projektu Startowego "hello world") o następującej strukturze:
```

touch global.css
```

> > Uwaga: Możesz użyc dowolnego edytora kodu, aby utworzyc poniższą strukturę plików i folderów, wedle własnych preferencji.

Author: ```suggestion
> Uwaga: Możesz użyć dowolnego edytora kodu, aby utworzyć poniższą strukturę plików i folderów, wedle własnych preferencji.
```

│       └── global.css
```

> 2. Zdefiniuj jakiś styl w pliku `global.css`:

Author: ```suggestion
2. Zdefiniuj dowolny styl w pliku `global.css`:
```

```css:title=src/styles/global.css
html {
  background-color: lavenderblush;
}
```

> > Uwaga: Umiejscowienie przykładowego pliku css w folderze `/src/styles/` jest dowolne.

Reviewer: ```suggestion
> Uwaga: Umiejscowienie przykładowego pliku css w folderze `/src/styles/` jest opcjonalne.
```

> #### ✋ Dodaj stylesheet do pliku `gatsby-browser.js`

Reviewer: ```suggestion
#### ✋ Dodaj arkusz styli do pliku `gatsby-browser.js`
```
tutaj zastanowiłbym się nad jakimś słowem, które lepiej tłumaczyłoby 'include', co myślisz o użyciu 'załącz' albo 'dołącz'?

Reviewer 2: Mi chyba bardziej pasuje 'dodaj' w tym wypadku.

> #### ✋ Dodaj arkusz styli do pliku `gatsby-browser.js`

Author: ```suggestion
#### ✋ Dodaj arkusz stylów do pliku `gatsby-browser.js`
```

├── gatsby-browser.js
```

> > 💡 Za co odpowiada `gatsby-browser.js`? Nie przejmuj się tym teraz, wiedz tylko, że `gatsby-browser.js` jest jednym z wielu specjalnych plików które Gatsby automatycznie wyszukuje i używa (o ile istnieją). W tym wypadku nazwa pliku **ma znaczenie**. Jeśli chcesz dowiedzieć się więcej na ten temat już teraz, sprawdź [Dokumentację](/docs/browser-apis/).

Author: ```suggestion
> 💡 Za co odpowiada `gatsby-browser.js`? Nie przejmuj się tym teraz, wiedz tylko, że `gatsby-browser.js` jest jednym z wielu specjalnych plików, które Gatsby automatycznie wyszukuje i używa (o ile istnieją). W tym wypadku nazwa pliku **ma znaczenie**. Jeśli chcesz dowiedzieć się więcej na ten temat już teraz, sprawdź [Dokumentację](/docs/browser-apis/).
```

> > 💡 Za co odpowiada `gatsby-browser.js`? Nie przejmuj się zbytnio tym na tym etapie, wiedz tylko, ze `gatsby-browser.js` jest jednym z wielu specjalnych plików które Gatsby automatycznie wyszukuje i uzywa (o ile istnieją). W tym wypadku nazwa pliku **ma znaczenie**. Jeśli chcesz dowiedzieć się więcej na ten temat już teraz, sprawdź [dokumentację](/docs/browser-apis/).

Reviewer: ```suggestion
> 💡 Za co odpowiada `gatsby-browser.js`? Nie przejmuj się tym teraz, wiedz tylko, że `gatsby-browser.js` jest jednym z wielu specjalnych plików które Gatsby automatycznie wyszukuje i używa (o ile istnieją). W tym wypadku nazwa pliku **ma znaczenie**. Jeśli chcesz dowiedzieć się więcej na ten temat już teraz, sprawdź [dokumentację](/docs/browser-apis/).
```
pozbyłem się powtórzonego 'tym' i dodałem polskie znaki

Reviewer 2: Mała uwaga/pytanie @lekterable @matborowiak. Co sądzicie aby  Dokumentacja pisać zawsze z wielkiej litery. W kilku miejsach widziałem Gatsby Docs lub Docs. Moglibśmy się tego trzymać wszędzie, co wy na to?

Author: ```suggestion
> 💡 Za co odpowiada `gatsby-browser.js`? Nie przejmuj się tym teraz, wiedz tylko, że `gatsby-browser.js` jest jednym z wielu specjalnych plików które Gatsby automatycznie wyszukuje i używa (o ile istnieją). W tym wypadku nazwa pliku **ma znaczenie**. Jeśli chcesz dowiedzieć się więcej na ten temat już teraz, sprawdź [Dokumentację](/docs/browser-apis/).
```

> 2. Zaimportuj swoje właśnie utworzone arkusze styli do pliku `gatsby-browser.js`:

Author: ```suggestion
2. Zaimportuj przed chwilą utworzone arkusze styli do pliku `gatsby-browser.js`:
```

> 2. Zaimportuj przed chwilą utworzone arkusze styli do pliku `gatsby-browser.js`:

Author: ```suggestion
2. Zaimportuj przed chwilą utworzone arkusze stylów do pliku `gatsby-browser.js`:
```

> 2. Zaimportuj swoje właśnie utworzone stylesheet'y do pliku `gatsby-browser.js`:

Reviewer: ```suggestion
2. Zaimportuj swoje właśnie utworzone arkusze styli do pliku `gatsby-browser.js`:
```

// require('./src/styles/global.css')
```

> > Uwaga: Zarówno syntax CommonJS (`require`) jak Moduł ES (`import`), zadziałają w tym wypadku. Jeśli nie jesteś pewien który z nich wybrać, my w większości wypadków używamy 'import'.

Reviewer 2: ```suggestion
> Składnia zarówno CommonJS (`require`), jak i ES Module (`import`), zadziałają w tym wypadku. Jeśli nie jesteś pewien który z nich wybrać, my w większości wypadków używamy 'import'.
```
Moim zdaniem ES Module to nazwa własna i trzeba zaostawić to w niezmienionej formie.

Reviewer: ```suggestion
> Uwaga: Zarówno składnia CommonJS (`require`), jak i ES Module (`import`), zadziała w tym wypadku. Jeśli nie jesteś pewien którą z nich wybrać, my w większości wypadków używamy 'import'.
```

> 3. Uruchom serwer deweloperski (development server):

Reviewer 3: [Dla spójności:](https://github.com/gatsbyjs/gatsby-pl/pull/6/files#diff-d20f31bd083ac5610b5bf5df061ad2dcR71)
```suggestion
3. Uruchom serwer developerski (development server):
```

Reviewer 2: Ja myślę, że powinniśmy zostać przy pisowni przez 'w'. Jak po Polsku to po Polsku :) co sądzicie?

Reviewer 3: Mi także bardziej pasuje pisownia przez `w`

Reviewer 2: Zaraz, ale zostawiłeś sugestię zmiany na 'v'? Czy czegoś nie rozumiem ?:)

Author: ```suggestion
3. Uruchom Serwer Deweloperski (development server):
```

Reviewer 3: @maciekgrzybek zostawiłem sugestię, bo gdzie indziej używamy pisowni przez `v`, więc dla spójności powinniśmy się trzymać jednej pisowni.

Nie zmienia to faktu, że osobiście preferuję pisownię przez `w` - proponuję razem zdecydować i dokonać zmianę w innych częściach naszych tłumaczeń. 

Schowam narazie moją pierwotną sugestię, bo wydaje mi się że chyba [wolimy pisownię](https://github.com/gatsbyjs/gatsby-pl/pull/8/files#diff-e4405b2d7facba43db450e0b7089654fR96) przez `w`  😄 .

Reviewer 2: Aaa spoko rozumiem :) wszystko jasne :)

Reviewer 2: @matborowiak może lepiej nie edytujmy komentarzy, tylko dodawajmy nowe? będziemy mieli jaśniejjszą sytuację i historie komentarzy krok po kroku.

Author: @maciekgrzybek jestem za. Chyba że literówki czy sugestje zmian w przypadku jakieś oczywistej wpadki :)

```shell
gatsby develop
```

> Jeli spojrzysz teraz na swój projekt w przeglądarce, powinieneś zobaczyć w starterze "hello world" zastosowane lawendowe tło:

Reviewer 3: ```suggestion
Jeśli spojrzysz teraz na swój projekt w przeglądarce, powinieneś zobaczyć w starterze "hello world" zastosowane lawendowe tło:
```

Author: ```suggestion
Jeśli spojrzysz teraz na swój projekt w przeglądarce, powinieneś zobaczyć w Starterze "hello world" zastosowane lawendowe tło:
```

![Lavender Hello World!](global-css.png)

> > Porada: Ta część tutorialu skupia się na najszybszym i prostoliniowym sposobie na rozpoczęcie stylowania strony Gatsby - bezpośrednim zaimportowaniu standardowych plików CSS uzywając `gatsby-browser.js`. Jednakl w większości wypadków, najelpszą metodą na zastosowanie stylów globalnych jest uzycie współdzielonego komponentu odpowiadającego za układ strony. [Sprawdź dokumentację](/docs/global-css/) aby dowiedzieć się więcej na ten temat.

Reviewer: ```suggestion
> Porada: Ta część poradnika skupia się na najszybszym i najprostszym sposobie na rozpoczęcie stylowania strony Gatsby - bezpośrednim zaimportowaniu standardowych plików CSS przy użyciu `gatsby-browser.js`. Jednak w większości przypadków, najlepszą metodą na zastosowanie stylów globalnych jest użycie współdzielonego komponentu odpowiadającego za układ strony. [Sprawdź dokumentację](/docs/global-css/) aby dowiedzieć się więcej na ten temat.
```

Author: ```suggestion
> Porada: Ta część poradnika skupia się na najszybszym i najprostszym sposobie na rozpoczęcie stylowania strony Gatsby - bezpośrednim zaimportowaniu standardowych plików CSS przy użyciu `gatsby-browser.js`. Jednak w większości przypadków, najlepszą metodą na zastosowanie Stylów Globalnych jest użycie współdzielonego komponentu odpowiadającego za układ strony. [Sprawdź Dokumentację](/docs/global-css/) aby dowiedzieć się więcej na ten temat.
```

> ## Użycie CSS o zasęgu kompnentu.

Reviewer: ```suggestion
## Użycie CSS o zasięgu komponentu.
```

Reviewer 2: ```suggestion
## Użycie CSS o zasęgu kompnentu
```

Author: ```suggestion
## Użycie CSS o zasięgu komponentu
```

> Do tej pory rozmawialiśmy o bardziej tradycyjnym podejściu do używania standardowych arkuszy stylów css. Teraz porozmawiamy o różnych metodach rozbicia CSS na niezależne moduły w celu zastosowania stylowania w sposób zorientowany na pracę z komponentami.

Author: ```suggestion
Do tej pory rozmawialiśmy o bardziej tradycyjnym podejściu do używania standardowych arkuszy styli CSS. Teraz porozmawiamy o różnych metodach rozbicia CSS na niezależne moduły w celu zastosowania stylowania w sposób zorientowany na pracę z komponentami.
```

### Moduły CSS

Zajmijmy się **Modułami CSS**. Cytując

> [stronę główną modółów CSS](https://github.com/css-modules/css-modules):

Reviewer 2: ```suggestion
[stronę główną modułów CSS](https://github.com/css-modules/css-modules):
```

Author: ```suggestion
[stronę główną Modułów CSS](https://github.com/css-modules/css-modules):
```

> **Moduł CSS** jest plikiem CSS, w którym wszystkie nazwy klas i nazwy animacji 
>  posiadają domyślny zasięg lokalny.

> Moduły CSS są bardzo popularne, ponieważ pozwalają normalnie pisać CSS, ale z większym bezpieczeństwem. Narzędzie automatycznie generuje unikalne nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw selektorów.

Reviewer: ```suggestion
Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły CSS, ale z większym bezpieczeństwem. Narzędzie automatycznie generuje unikalne nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw selektorów.
```

Reviewer 2: ```suggestion
Moduły CSS są bardzo popularne, ponieważ pozwalają normalnie zwykły, lecz bardziej bezpieczny kod  CSS.  Narzędzie automatycznie generuje unikalne nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw selektorów.
```

Reviewer: ```suggestion
Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły, lecz bezpieczniejszy kod CSS. Narzędzie automatycznie generuje unikalne nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw selektorów.
```
^ zjadłeś słowo 'pisać', ale podoba mi się pomysł. może taka ulepszona wersja?

Author: ```suggestion
Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły, lecz bardziej bezpieczny kod  CSS. Narzędzie automatycznie generuje unikalne nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw Selektorów.
```

> Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły, lecz bardziej bezpieczny kod  CSS. Narzędzie automatycznie generuje unikalne nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw Selektorów.

Author: ```suggestion
Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły, lecz bardziej bezpieczny kod  CSS. Narzędzie automatycznie generuje unikatowe nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw Selektorów.
```

> Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły, lecz bardziej bezpieczny kod  CSS. Narzędzie automatycznie generuje unikatowe nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw Selektorów.

Reviewer: ```suggestion
Moduły CSS są bardzo popularne, ponieważ pozwalają pisać zwykły, lecz bardziej bezpieczny kod CSS. Narzędzie automatycznie generuje unikatowe nazwy klas i animacji, więc nie musisz się martwić o kolizje nazw Selektorów.
```

> Gatsby działa od razu z modułami CSS. Ten sposób jest wysoce zalecany dla osób, które dopiero zaczynają budować strony z Gatsby (i ogólnie z React'em).

Reviewer 2: ```suggestion
Gatsby działa domyślnie z modułami CSS. Ten sposób jest wysoce zalecany dla osób, które dopiero zaczynają budować strony z Gatsby (i ogólnie z React'em).
```

Reviewer: ```suggestion
Gatsby ma wbudowane wsparcie dla modułów CSS. Ten sposób jest wysoce zalecany dla osób, które dopiero zaczynają budować strony z Gatsby (i ogólnie z React'em).
```
a może coś takiego?

Author: ```suggestion
Gatsby ma wbudowane wsparcie dla Modułów CSS. Ten sposób jest wysoce zalecany dla osób, które dopiero zaczynają budować strony z Gatsby (i ogólnie z React'em).
```

> Gatsby działa od razu z modułami CSS. Ten sposób jest wysoce zalecany dla osób, które dopiero zaczynają budować strony z Gatsby (i ogólnie z React'em).

Reviewer: ```suggestion
Gatsby ma wbudowane wsparcie dla modułów CSS. Ten sposób jest wysoce zalecany dla osób, które dopiero zaczynają budować strony z Gatsby (i ogólnie z React'em).
```
a może coś takiego?

#### ✋ Zbuduj nową stronę używając Modułów CSS

W tej sekcji będziesz tworzyć komponent nowej strony oraz ostylujesz ten komponent używając Modułu CSS.

> Po pierwsze, utwórz nowy komponent 'Container'.

Author: ```suggestion
Po pierwsze, utwórz nowy Komponent `Container`.
```

> 1. Utwórz nowy katalog w `src/components`, a następnie w tym nowym katalogu utwórz plik o nazwie `container.js` i wklej następujące elementy:

Reviewer 2: ```suggestion
1. Utwórz nowy katalog w `src/components`, a następnie w tym nowym katalogu utwórz plik o nazwie `container.js` i wklej następujący kod:
```

)
```

> Zauważ, że zaimportowałeś plik modułu css o nazwie `container.module.css`. Utwórzmy teraz ten plik.

Author: ```suggestion
Zauważ, że zaimportowałeś plik Modułu CSS o nazwie `container.module.css`. Utwórzmy teraz ten plik.
```

> Zobaczysz, że zaimportowałeś plik modułu css o nazwie `container.module.css`. Utwórzmy teraz ten plik.

Reviewer 2: ```suggestion
Zauważ, jak zaimportowałeś plik modułu css o nazwie `container.module.css`. Teraz utwórzmy ten plik.
```

Reviewer: ```suggestion
Zauważ, że zaimportowałeś plik modułu css o nazwie `container.module.css`. Utwórzmy teraz ten plik.
```
łącząc po części obie propozycje, najbardziej przemawia do mnie coś takiego

> 2. W tym samym katalogu (`src/components`), utórz plik `container.module.css` i kopiuj/wklej następujący kod:

Reviewer 3: ```suggestion
2. W tym samym katalogu (`src/components`), utwórz plik `container.module.css` i kopiuj/wklej następujący kod:
```

}
```

Zauważ, że nazwa pliku kończy się na `.module.css` zamiast zwykłego `.css`. W ten sposób informujesz Gatsby, że ten plik CSS powinien być przetwarzany jako Moduł CSS, zamiast zwykłego CSS.

> 3. Stwórz komponent nowej strony poprzez utworzenie pliku w:

Author: ```suggestion
3. Stwórz Komponent nowej strony poprzez utworzenie pliku w:
```

)
```

Teraz, gdy odwiedzisz `http://localhost:8000/about-css-modules/`, Twoja strona powinna wyglądać mniej więcej tak:

![Strona ze stylami Modułów CSS](css-modules-basic.png)

#### ✋ Ostyluj komponent używając Modułów CSS

> W tej sekcji utworzysz listę osób o imionach, awatarach i krótkich biografiach łacińskich. Utworzysz także Komponent `<User />' i nadasz mu Styl za pomocą Modułu CSS.

Author: ```suggestion
W tej sekcji utworzysz listę osób o imionach, awatarach i krótkich biografiach łacińskich. Utworzysz także Komponent `<User />' i nadasz mu styl za pomocą Modułu CSS.
```

> W tej sekcji utworzysz listę osób o imionach, awatarach i krótkich biografiach łacińskich. Utworzysz także komponent `<User />', i nadasz mu styl za pomocą Modułu CSS.

Reviewer: ```suggestion
W tej sekcji utworzysz listę osób o imionach, awatarach i krótkich biografiach łacińskich. Utworzysz także komponent `<User />' i nadasz mu styl za pomocą Modułu CSS.
```

Author: ```suggestion
W tej sekcji utworzysz listę osób o imionach, awatarach i krótkich biografiach łacińskich. Utworzysz także Komponent `<User />' i nadasz mu Styl za pomocą Modułu CSS.
```

1. Utwórz plik dla CSS w `src/pages/about-css-modules.module.css`.

> 2. Wklej poniższy kod do nowo utworzonego pliku pliku:

Reviewer 2: ```suggestion
2. Wklej poniższy kod do nowo utworzonego pliku:
```

> 2. Wklej poniższy kod do nowo utworzonego pliku pliku:

Reviewer 3: ```suggestion
2. Wklej poniższy kod do nowo utworzonego pliku:
```

}
```

> 3. Zaimportuj nowy plik `src/pages/about-css-modules.module.css` do strony `about-css-modules.js`, którą wcześniej utworzyłeś, poprzez edytowanie pierwszych kilku linijek kodu w pliku, w ten sposób:

Reviewer 2: ```suggestion
3. Zaimportuj nowy plik `src/pages/about-css-modules.module.css` do strony `about-css-modules.js`, którą wcześniej utworzyłeś, poprzez edytowanie pierwszych kilku linijek kodu, w ten sposób:
```

console.log(styles)
```

> Kod `console.log(styles)` wyświetli wykonany import, a więc zobaczysz wynik pochodzący z twojego przetworzonego pliku `./about-css-modules.module.css`. Jeśli otworzysz konsolę deweloperską (np. nażędzi deweloperskich Firefox'a lub Chrome) w swojej przeglącarce, oto co zobaczysz:

Reviewer: ```suggestion
Kod `console.log(styles)` wyświetli wykonany import, a więc zobaczysz wynik pochodzący z twojego przetworzonego pliku `./about-css-modules.module.css`. Jeśli otworzysz konsolę deweloperską (korzystając np. z narzędzi deweloperskich Firefox'a lub Chrome) w swojej przeglącarce, oto co zobaczysz:
```

Author: ```suggestion
Kod `console.log(styles)` wyświetli wykonany import, a więc zobaczysz wynik pochodzący z twojego przetworzonego pliku `./about-css-modules.module.css`. Jeśli otworzysz Konsolę Deweloperską (korzystając np. z narzędzi deweloperskich Firefox'a lub Chrome) w swojej przeglądarce, oto co zobaczysz:
```

> Kod `console.log(styles)` wyświetli wykonany import, a więc zobaczysz wynik pochodzący z twojego przetworzonego pliku `./about-css-modules.module.css`. Jeśli otworzysz konsolę deweloperską (np. nażędzi deweloperskich Firefox'a lub Chrome) w swojej przeglącarce, oto co zobaczysz:

Reviewer 2: ```suggestion
Kod `console.log(styles)` wyświetli zaimportowany moduł, dzięki czemu zobaczysz swój przetworzony plik `./about-css-modules.module.css`. Jeśli otworzysz konsolę deweloperską (korzystając np. z narzędzi deweloperskich Firefox'a lub Chrome) w swojej przeglądarce, oto co zobaczysz:
```

> Kod `console.log(styles)` wyświetli wykonany import, a więc zobaczysz wynik pochodzący z twojego przetworzonego pliku `./about-css-modules.module.css`. Jeśli otworzysz konsolę deweloperską (np. nażędzi deweloperskich Firefox'a lub Chrome) w swojej przeglącarce, oto co zobaczysz:

Reviewer 3: Zgadzam się z sugestią @lekterable, wyłapałem jeszcze jedną literówkę: 
```suggestion
Kod `console.log(styles)` wyświetli wykonany import, a więc zobaczysz wynik pochodzący z twojego przetworzonego pliku `./about-css-modules.module.css`. Jeśli otworzysz konsolę deweloperską (np. nażędzi deweloperskich Firefox'a lub Chrome) w swojej przeglądarce, oto co zobaczysz:
```

![Rezultat zaimportowania Modułu CSS w konsoli](css-modules-console.png)

> Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym do długiego string'a - np. 'avatar' wkazuje do `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Jest zagwarantowane, że nazwy te będą unikatowe na całej Twojej stronie. I ponieważ dlatego, że muszisz zaimportować je by użyć klas, nie ma nigdy wątpliwości, gdzie jaki CSS został użyty.

Reviewer: ```suggestion
Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym na długiego string'a - np. 'avatar' wkazuje na `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Jest zagwarantowane, że nazwy te będą unikatowe na całej Twojej stronie. I ponieważ musisz zaimportować je, by użyć klas, nie ma nigdy wątpliwości, gdzie jaki CSS został użyty.
```

Author: ```suggestion
Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym na długi String - np. 'avatar' wskazuje na `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Jest zagwarantowane, że nazwy te będą unikatowe na całej Twojej stronie. I ponieważ musisz zaimportować je, by użyć klas, nie ma nigdy wątpliwości, gdzie jaki CSS został użyty.
```

Author: @lekterable @maciekgrzybek , przyuważyłem w słowniczku że String to "ciąg znaków", nie wydaje mi się że tłumaczenie tak pierwotnych pojęć dla programowania jak typy danych String czy Integer, Undefined lub Null jest dobrym pomysłem... Co sądzicie na ten temat?

Reviewer: sposób w jaki to użyłeś tutaj, tzn. z dużej litery i bez odmiany wygląda dla mnie bardzo spoko więc ode mnie 👍

> Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym do długiego string'a - np. 'avatar' wkazuje do `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Jest zagwarantowane, że nazwy te będą unikatowe na całej Twojej stronie. I ponieważ dlatego, że muszisz zaimportować je by użyć klas, nie ma nigdy wątpliwości, gdzie jaki CSS został użyty.

Reviewer 2: ```suggestion
Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym do długi łańcuch znaków - np. 'avatar' wskazuje na `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Gwarantuje to, że nazwy te będą unikatowe w skali całej strony. Ponadto fakt, iż musisz zaimportować dany moduł aby użyć klas, nie pozostawia żadnych wątpliwości, gdzie jaki kod CSS został użyty.
```

> Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym do długiego string'a - np. 'avatar' wkazuje do `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Jest zagwarantowane, że nazwy te będą unikatowe na całej Twojej stronie. I ponieważ dlatego, że muszisz zaimportować je by użyć klas, nie ma nigdy wątpliwości, gdzie jaki CSS został użyty.

Reviewer 3: zgadzam się z @lekterable , złapałem jeszcze jedną literówkę:
```suggestion
Jeśli porównasz to ze swoim plikiem CSS, zauważysz, że każda z klas jest teraz kluczem w zaimportowanym obiekcie wskazującym do długiego string'a - np. 'avatar' wskazuje do `src-pages----about-css-modules-module---avatar---2lRF7`. Tak wyglądają nazwy klas wygenerowane przez Moduł CSS. Jest zagwarantowane, że nazwy te będą unikatowe na całej Twojej stronie. I ponieważ dlatego, że muszisz zaimportować je by użyć klas, nie ma nigdy wątpliwości, gdzie jaki CSS został użyty.
```
Nie jestem też pewien użycia słowa `string` -- w "[słowniczku](https://github.com/gatsbyjs/gatsby-pl/blob/master/style-guide.md#s%C5%82owniczek)" korzystamy z tłumaczenia "Łańcuch znaków"

4. Stwórz komponent `User`.

> Stwórz nowy komponent `<User />` w lini w komponencie strony `about-css-modules.js`. 

Reviewer: ```suggestion
Stwórz nowy komponent `<User />` w linii w komponencie strony `about-css-modules.js`. 
```

Reviewer 2: Te w lini jakoś nie pasuje to w ogóle. Może np  'bezpośrednio w komponencie' ?

Author: ```suggestion
Stwórz nowy komponent `<User />` bezpośrednio w kodzie komponentu strony `about-css-modules.js`. 
```

)
```

> > Wskazówka: Generalnie, jeśli użyjesz komponentu w kilku różnych miejscach na stronie, powinien on zawierać się we własnym pliku module w folderze 'components'. Ale gdy jest użyty tylko w jednym pliku, napisz go "w linii".

Reviewer: ```suggestion
> Wskazówka: Generalnie, jeśli użyjesz komponentu w kilku różnych miejscach na stronie, powinien on znajdować się we własnym pliku module w folderze 'components'. Ale gdy jest użyty tylko w jednym pliku, napisz go "w linii".
```

Reviewer 2: Tak samo tutaj.

Reviewer 2: ```suggestion
> Wskazówka: Generalnie, jeśli użyjesz komponentu w kilku różnych miejscach na stronie, powinien on zawierać się we własnym pliku/module w folderze 'components'. Ale gdy jest użyty tylko w jednym pliku, napisz go "w linii".
```

Author: ```suggestion
> Wskazówka: Generalnie, jeśli użyjesz komponentu w kilku różnych miejscach na stronie, powinien on znajdować się we własnym pliku-module w folderze 'components'. Ale gdy jest użyty tylko w jednym pliku, napisz go bezpośrednio w kodzie.
```

> > Wskazówka: Generalnie, jeśli użyjesz komponentu w kilku różnych miejscach na stronie, powinien on zawierać się we własnym pliku module w folderze 'components'. Ale gdy jest użyty tylko w jednym pliku, napisz go "w linii".

Reviewer 3: Można też zrobić to tak (spójnie z naszym tłumaczeniem słowa [Moduł CSS](https://github.com/gatsbyjs/gatsby-pl/pull/7/files#diff-56e3fb760cff8a5eadfc58a75b142c88R140)):
```suggestion
> Wskazówka: Generalnie, jeśli zamierzasz używać komponentu w kilku różnych miejscach na stronie, powinien on znajdować się we własnym Module CSS w folderze `components`. Ale gdy jest użyty tylko w jednym pliku, napisz go bezpośrednio w kodzie.
```
Dajcie znać, co myślicie. @lekterable @matborowiak

Author: Podoba mi sie Twój poczatek zdania. Co robimy kiedy sugestie sa 'outdated' ???

Reviewer: ```suggestion
> Wskazówka: Generalnie, jeśli zamierzasz używać komponentu w kilku różnych miejscach na stronie, powinien znajdować się on we własnym Module CSS w folderze `components`. Ale jeśli jest użyty tylko w jednym pliku, umieść go bezpośrednio w kodzie.
```

hm?

Reviewer: @matborowiak myślę, że najlepiej będzie jeśli zedytujesz to ręcznie i wkleisz odnośnik z linią kodu przed rozwiązaniem konwersacji

Reviewer: @matborowiak albo można otworzyć nową sugestie dla ułatwienia: https://github.com/gatsbyjs/gatsby-pl/pull/7/files#r341819439

> > Wskazówka: Generalnie, jeśli użyjesz komponentu w kilku różnych miejscach na stronie, powinien on znajdować się we własnym pliku-module w folderze 'components'. Ale gdy jest użyty tylko w jednym pliku, napisz go bezpośrednio w kodzie.

Reviewer: ```suggestion
> Wskazówka: Generalnie, jeśli zamierzasz używać komponentu w kilku różnych miejscach na stronie, powinien znajdować się on we własnym Module CSS w folderze `components`. Ale jeśli jest użyty tylko w jednym pliku, możesz umieścić go bezpośrednio w kodzie.
```

Author: Mysle ze ostatnie zdanie powinno zawierać sugestie że jest to opcjonalne... Czasami piszemy przecież komponenty na pojedynczy użytek w odrebnych plikach, bo nie wiadomo kiedy się przyda. Także nie chciałbym wprowadzić kogoś w błąd gdyby omyłkowo zrozumiał że jak używasz komponentu raz, to zawsze ładuj go bezpośrednio do pliku w którym zostanie użyty

Reviewer: myślisz, że zamiana na 'możesz umieścić go bezpośrednio w kodzie' wystarczy, czy potrzebujemy czegoś bardziej dosadnego?

Author: myślę że słówko 'możesz' załatwia sprawę!

Reviewer 2: Tutaj chyba nie zmieniliście tego co? Bo tylko tego nam brakuje do merge z tego co widzę

Reviewer: outdated, dałem nową sugestie: https://github.com/gatsbyjs/gatsby-pl/pull/7/files#r341823142

> > Wskazówka: Wskazówka: Generalnie, jeśli zamierzasz używać komponentu w kilku różnych miejscach na stronie, powinien znajdować się on we własnym Module CSS w folderze `components`. Ale jeśli jest użyty tylko w jednym pliku, umieść go bezpośrednio w kodzie.

Reviewer: ```suggestion
> Wskazówka: Generalnie, jeśli zamierzasz używać komponentu w kilku różnych miejscach na stronie, powinien znajdować się on we własnym Module CSS w folderze `components`. Ale jeśli jest użyty tylko w jednym pliku, możesz umieścić go bezpośrednio w kodzie.
```

Skończona strona powinna wyglądać w taki sposób:

![Strona listy użytkowników z Modułami CSS](css-modules-userlist.png)

### CSS-in-JS

> CSS-in-JS jest metodą stylowania zorientowaną na komponenty. Najogólniej mówiąc, jest to schemat w którym [CSS jest wkomponowany w linii używając JavaScript](https://reactjs.org/docs/faq-styling.html#what-is-css-in-js).

Reviewer 2: ```suggestion
CSS-in-JS jest metodą stylowania zorientowaną na komponenty. Najogólniej mówiąc, jest to schemat w którym [CSS jest napisany bezpośrednio w pliku JavaScript](https://reactjs.org/docs/faq-styling.html#what-is-css-in-js).
```

#### Używanie CSS-in-JS z Gatsby

> Jest wiele różnych bibliotek CSS-in-JS i wiele z nich już posiada wtyczki do Gatsby. Nie będziemy omawiać przykładu CSS-in-JS w tym początkowym samouczku, ale zachęcamy do [zbadania](/docs/styling/) co ten ekosystem ma do zaoferowania. Mamy mini-samouczki dla dwóch bibliotek, a konkretnie, [Emotion](/docs/emotion/) oraz [Styled Components](/docs/styled-components/).

Reviewer 2: ```suggestion
Jest wiele różnych bibliotek CSS-in-JS i wiele z nich posiada juź wtyczki do Gatsby. Nie będziemy omawiać przykładu CSS-in-JS w tym początkowym samouczku, ale zachęcamy Cię do [zbadania](/docs/styling/) co ten ekosystem ma do zaoferowania. Mamy mini-samouczki dla dwóch bibliotek, a konkretnie, [Emotion](/docs/emotion/) oraz [Styled Components](/docs/styled-components/).
```

Author: ```suggestion
Jest wiele różnych bibliotek CSS-in-JS i wiele z nich posiada już wtyczki do Gatsby. Nie będziemy omawiać przykładu CSS-in-JS w tym początkowym samouczku, ale zachęcamy Cię do [zbadania](/docs/styling/) co ten ekosystem ma do zaoferowania. Mamy mini-samouczki dla dwóch bibliotek, a konkretnie, [Emotion](/docs/emotion/) oraz [Styled Components](/docs/styled-components/).
```

> Jest wiele różnych bibliotek CSS-in-JS i wiele z nich posiada już wtyczki do Gatsby. Nie będziemy omawiać przykładu CSS-in-JS w tym początkowym samouczku, ale zachęcamy Cię do [zbadania](/docs/styling/) co ten ekosystem ma do zaoferowania. Mamy mini-samouczki dla dwóch bibliotek, a konkretnie, [Emotion](/docs/emotion/) oraz [Styled Components](/docs/styled-components/).

Author: ```suggestion
Jest wiele różnych bibliotek CSS-in-JS i wiele z nich posiada już wtyczki do Gatsby. Nie będziemy omawiać przykładu CSS-in-JS w tym początkowym samouczku, ale zachęcamy Cię do [zbadania](/docs/styling/) co ten ekosystem ma do zaoferowania. Mamy mini-poradniki dla dwóch bibliotek, a konkretnie, [Emotion](/docs/emotion/) oraz [Styled Components](/docs/styled-components/).
```

#### Zalecane lektury na temat CSS-in-JS

> Jeśli jesteś zainteresowany dalszą lekturą, sprawdź [Christopher "vjeux" Chedeau's 2014 presentation that sparked this movement](https://speakerdeck.com/vjeux/react-css-in-js), a także [jeden z nowszych wpisów Mark'a Dalgleish "A Unified Styling Language"](https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660).

Reviewer: ```suggestion
Jeśli jesteś zainteresowany dalszą lekturą, sprawdź [Christopher "vjeux" Chedeau's 2014 presentation that sparked this movement](https://speakerdeck.com/vjeux/react-css-in-js), a także [jeden z nowszych wpisów Mark'a Dalgleish'a "A Unified Styling Language"](https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660).
```

Reviewer 3: Odnośnie odmiany angielskich imion/nazwisk - http://forum.mlingua.pl/showpost.php?s=2e24c37ac60e978c998c9d152266469a&p=167095&postcount=6
Przetłumaczyłem też ten fragment:
```suggestion
Jeśli jesteś zainteresowany dalszą lekturą, sprawdź [prezentację Christophera "vjeux" Chedeau'sa z roku 2014, która zapoczątkowała ten ruch](https://speakerdeck.com/vjeux/react-css-in-js), a także [jeden z nowszych wpisów Marka Dalgleish'a - "A Unified Styling Language"](https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660).
```

> Jeśli jesteś zainteresowany dalszą lekturą, sprawdź [Christopher "vjeux" Chedeau's 2014 presentation that sparked this movement](https://speakerdeck.com/vjeux/react-css-in-js), a także [jeden z nowszych wpisów Mark'a Dalgleish'a "A Unified Styling Language"](https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660).

Reviewer 3: Wstawiam jeszcze raz, bo poprzedni [komentarz](https://github.com/gatsbyjs/gatsby-pl/pull/7#discussion_r341816850) został oznaczony jako 'outdated'

Odnośnie odmiany angielskich imion/nazwisk, znalazłem taką pomoc: http://forum.mlingua.pl/showpost.php?s=2e24c37ac60e978c998c9d152266469a&p=167095&postcount=6
Przetłumaczyłem też ten fragment:
```suggestion
Jeśli jesteś zainteresowany dalszą lekturą, sprawdź [prezentację Christophera "vjeux" Chedeau'sa z roku 2014, która zapoczątkowała ten ruch](https://speakerdeck.com/vjeux/react-css-in-js), a także [jeden z nowszych wpisów Marka Dalgleish'a - "A Unified Styling Language"](https://medium.com/seek-blog/a-unified-styling-language-d0c208de2660).
```

### Inne opcje CSS

Gatsby wspiera prawie każdą możliwą opcję stylowania (jeśli wciąż nie ma wtyczki do twojego ulubionego sposobu na użycie CSS, [prosimy o twój wkład!](/contributing/how-to-contribute/))

- [Typography.js](/packages/gatsby-plugin-typography/)
- [Sass](/packages/gatsby-plugin-sass/)
- [JSS](/packages/gatsby-plugin-jss/)
- [Stylus](/packages/gatsby-plugin-stylus/)
- [PostCSS](/packages/gatsby-plugin-postcss/)

i więcej!

## Czego nauczysz się następnie?

> Przejdź teraz do [części trzeciej samouczka](/tutorial/part-three/), gdzie dowiesz się informacji na temat wtyczek Gatsby i komponentów Układu Stron.

Reviewer: ```suggestion
Przejdź teraz do [części trzeciej samouczka](/tutorial/part-three/), gdzie dowiesz się więcej na temat wtyczek Gatsby i komponentów Układu Stron.
```

> Przejdź teraz do [części trzeciej samouczka](/tutorial/part-three/), gdzie dowiesz się więcej na temat wtyczek Gatsby i komponentów Układu Stron.

Author: ```suggestion
Przejdź teraz do [części trzeciej poradnika](/tutorial/part-three/), gdzie dowiesz się więcej na temat wtyczek Gatsby i komponentów Układu Stron.
```
